### PR TITLE
Changes to support the compiler portion of P1152R4 "deprecating volatile"

### PR DESCRIPTION
**Pull request by CaseyCarter**
* Add the new warning C5215 "'%s' a function parameter with volatile qualified type is deprecated in C++20" to libc++'s "suppress deprecation warnings" macro and update llvm submodule reference.
* Skip libc++ test `containers/sequences/deque/deque.modifiers/insert_iter_iter.pass.cpp` after seeing one timeout too many in Contest logs.
* Suppress C5215 in `tr1/include/tfuns.h` which uses volatile function parameters.
* Update skips for two libc++ tests that changed suffixes from `.sh.cpp` to `.pass.cpp`.
* Enable C5214-C5217 (which are off-by-default) in libc++ and std test suites

[This is a dual of internal MSVC-PR-239335.]

### CODE REVIEW

File: tests/tr1/include/tfuns.h [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 // common header for functional?.cpp
 
+#pragma warning(push)
+#pragma warning(disable : 5215) // '%s' a function parameter with volatile qualified type is deprecated in C++20
```
Comment (CaseyCarter) [inline tests/tr1/include/tfuns.h:7]: The warning push/disable/pop is the only function change to this header; I made a non-functional drive-by fix to some weird formatting on lines 135-220.

File: stl/inc/optional [hunk 2]
```diff
@@ -394,7 +394,7 @@ public:
     }
 
     template <class _Ty2>
-    _NODISCARD constexpr _Ty value_or(_Ty2&& _Right) const& {
+    _NODISCARD constexpr remove_cv_t<_Ty> value_or(_Ty2&& _Right) const& {
```
Comment (CaseyCarter) [inline stl/inc/optional:397]: I've submitted an LWG issue (but not yet been assigned a number) to make the changes to both `value_or` overloads Standard.

File: stl/inc/type_traits [hunk 3]
```diff
@@ -1660,13 +1660,14 @@ struct _Invoker_ret<_Unforced, false> { // selected for _Rx being _Unforced
...
 
 // TYPE TRAITS FOR invoke()
-template <class _To>
-void _Implicitly_convert_to(_To) noexcept; // not defined
-
 #pragma warning(push)
 #pragma warning(disable : 4242) // 'identifier': conversion from '_From' to '_To', possible loss of data (/Wall)
 #pragma warning(disable : 4244) // 'argument': conversion from '_From' to '_To', possible loss of data
 #pragma warning(disable : 4365) // 'argument': conversion from '_From' to '_To', signed/unsigned mismatch (/Wall)
+#pragma warning(disable : 5215) // '%s' a function parameter with a volatile qualified type is deprecated in C++20
```
Comment (StephanTLavavej) [inline stl/inc/type_traits:1667]: I find it somewhat interesting that you don't need to suppress Clang `-Wdeprecated-volatile` here. But if there are no warnings in our test suite, I'm happy.

File: tests/utils/stl/test/format.py [hunk 4]
```diff
@@ -286,3 +286,9 @@ def getTestsInDirectory(self, testSuite, path_in_suite,
                             litConfig, localConfig, test_class=LibcxxTest):
         return super().getTestsInDirectory(testSuite, path_in_suite, litConfig,
                                            localConfig, test_class)
+
+    def addCompilerFlags(self, *args):
```
Comment (CaseyCarter) [inline tests/utils/stl/test/format.py:290]: You had one job!
```suggestion
    def addCompileFlags(self, *args):
```

File: stl/inc/type_traits (hunk 3 above)
Comment (CaseyCarter) [reply stl/inc/type_traits:1667]: We probably should; this will be in some future set of "suppress clang warnings that don't show up because system header" 🤷‍♂ .
